DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -56,12 +61,92 @@ function Set-TargetResource
 
         [uInt32]$SubnetMask,
 
-        [ValidateSet("IPv4", "IPv6")]
-        [String]$AddressFamily = "IPv4"
+        [ValidateSet('IPv4', 'IPv6')]
+        [String]$AddressFamily = 'IPv4'
     )
 
-    Test-Properties @PSBoundParameters -Apply
-}
+    try
+    {
+        Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+            'Applying the IP Address...'
+            ) -join '' )
+
+        #Should not need to validate the settings because Test-TargetResource already did this
+
+        # Use $AddressFamily to select the IPv4 or IPv6 destination prefix
+        $DestinationPrefix = '0.0.0.0/0'
+        if($AddressFamily -eq 'IPv6')
+        {
+            $DestinationPrefix = '::/0'
+        }
+
+        # Get all the default routes - this has to be done in case the IP Address is
+        # beng Removed
+        $defaultRoutes = @(Get-NetRoute `
+            -InterfaceAlias $InterfaceAlias `
+            -AddressFamily $AddressFamily `
+            -ErrorAction Stop).Where( { $_.DestinationPrefix -eq $DestinationPrefix } )
+
+        # Remove any default routes on the specified interface -- it is important to do
+        # this *before* removing the IP address, particularly in the case where the IP
+        # address was auto-configured by DHCP
+        if($defaultRoutes)
+        {
+            foreach ($defaultRoute in $defaultRoutes) {
+                Remove-NetRoute `
+                    -DestinationPrefix $defaultRoute.DestinationPrefix `
+                    -NextHop $defaultRoute.NextHop `
+                    -InterfaceIndex $defaultRoute.InterfaceIndex `
+                    -AddressFamily $defaultRoute.AddressFamily `
+                    -Confirm:$false -ErrorAction Stop
+            }
+        }
+
+        # Get the current IP Address based on the parameters given.
+        $currentIPs = @(Get-NetIPAddress `
+            -InterfaceAlias $InterfaceAlias `
+            -AddressFamily $AddressFamily `
+            -ErrorAction Stop)
+
+        # Remove any IP addresses on the specified interface
+        if($currentIPs)
+        {
+            foreach ($CurrentIP in $CurrentIPs) {
+                Remove-NetIPAddress `
+                    -IPAddress $CurrentIP.IPAddress `
+                    -InterfaceIndex $CurrentIP.InterfaceIndex `
+                    -AddressFamily $CurrentIP.AddressFamily `
+                    -Confirm:$false -ErrorAction Stop
+            }
+        }
+
+        # Build parameter hash table
+        $Parameters = @{
+            IPAddress = $IPAddress
+            PrefixLength = $SubnetMask
+            InterfaceAlias = $InterfaceAlias
+        }
+
+        # Apply the specified IP configuration
+        $null = New-NetIPAddress @Parameters -ErrorAction Stop
+
+        # Make the connection profile private
+        Get-NetConnectionProfile -InterfaceAlias $InterfaceAlias | `
+            Set-NetConnectionProfile -NetworkCategory Private -ErrorAction SilentlyContinue
```

Comment:
I'll remove it as I think #19 should be implemented reasonably quickly (I'm happy to put code in for that later on this week once this PR is done).